### PR TITLE
gateio: Remove `listOfAssetsCurrencyPairEnabledFor` method and optimise websocket asset processing

### PR DESCRIPTION
**Pull request by shazbert**
# PR Description

* Remove `listOfAssetsCurrencyPairEnabledFor` as it had performance overhead when processing an update
* Cleans up code
* ~~Adds helper method for currency pair `Compact` to just derive a string rep without the delimiter~~ SCOTT SAID NO 🙅 with good reason

Fixes # (issue)

## Type of change

Please delete options that are not relevant and add an `x` in `[]` as item is complete.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How has this been tested

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration and
also consider improving test coverage whilst working on a certain feature or package.

- [ ] go test ./... -race
- [ ] golangci-lint run
- [ ] Test X

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally and on Github Actions with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by gloriousCode — CHANGES_REQUESTED** [review]
Looks good! Just one minor change

File: exchanges/gateio/gateio_websocket.go [hunk 1]
```diff
@@ -160,37 +162,28 @@ func (g *Gateio) WsHandleSpotData(_ context.Context, respRaw []byte) error {
...
-		marginTicker.AssetType = asset.Margin
-		g.Websocket.DataHandler <- &marginTicker
-	}
-	if assetPairEnabled[asset.CrossMargin] {
-		crossMarginTicker := tickerPrice
-		crossMarginTicker.AssetType = asset.CrossMargin
-		g.Websocket.DataHandler <- &crossMarginTicker
-	}
+	out := make([]ticker.Price, 0, len(standardMarginAssetTypes))
+	for _, a := range standardMarginAssetTypes {
```
[the remark this answers is not in the document]
Comment (gloriousCode) [reply exchanges/gateio/gateio_websocket.go:169]: @gloriousCode this code sucks, [boo](https://www.youtube.com/watch?v=2Kpb8eu1pEY)

**Review by gloriousCode — APPROVED** [review]
tACK! Thanks for highlighting my code sucked

File: exchanges/gateio/gateio_websocket.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (thrasher-) [reply exchanges/gateio/gateio_websocket.go:169]: It was worth it seeing glorious and shazbert butts!

**Review by thrasher- — APPROVED** [review]
tACK 🐎

**Pull request merged**